### PR TITLE
Use puppet reload instead of puppet restart

### PR DESCRIPTION
**Pull request by sathieu**
This avoid killing the current puppet agent run

### CODE REVIEW

**Comment by sathieu** [comment]
Tested on Debian with and without puppetlabs repo. Tested on CentOS6 with puppetlabs repos

**Comment by mmoll** [comment]
At least FreeBSD has no "reload" for the puppet service, but I get the problem...

File: manifests/agent/service.pp [hunk 1]
```diff
@@ -4,10 +4,12 @@
...
-        ensure    => running,
-        name      => $puppet::service_name,
-        hasstatus => true,
-        enable    => true,
+        ensure     => running,
+        name       => $puppet::service_name,
+        hasstatus  => true,
+        hasrestart => true,
+        enable     => true,
+        restart    => '/usr/sbin/service puppet reload',
```
Comment (ekohl) [inline manifests/agent/service.pp]: I don't think this works on windows. Agent has a decent support matrix so maybe this should be in params.pp and undef for most OS-es.

**Comment by sathieu** [comment]
Here's an updated patch.

File: manifests/params.pp [hunk 2]
```diff
@@ -191,6 +191,11 @@
 
   # Puppet service name
   $service_name = 'puppet'
+  $agent_restart_command = $::osfamily {
```
Comment (mmoll) [inline manifests/params.pp]: missing case? :)

File: manifests/agent/service.pp [hunk 3]
```diff
@@ -4,10 +4,12 @@
   case $::puppet::runmode {
     'service': {
       service {'puppet':
-        ensure    => running,
-        name      => $puppet::service_name,
-        hasstatus => true,
-        enable    => true,
+        ensure     => running,
+        name       => $puppet::service_name,
+        hasstatus  => true,
```
Comment (ekohl) [inline manifests/agent/service.pp]: I suppose we can drop this now we only support puppet 2.7.0+

**Comment by mmoll** [comment]
works for me on FreeBSD

**Comment by ekohl** [comment]
Code looks good, :+1:

**Comment by mmoll** [comment]
merged as 8846c84, thanks @sathieu!

**Pull request closed without merging**